fendermint/fendermint/testing/contract-test/tests/staking/machine.rs

```diff
@@ -361,9 +363,11 @@ impl StateMachine for StakingMachine {
                     result.expect("checkpoint submission should succeed");
                 }
             }
-            StakingCommand::Join(_, value, _) => {
+            StakingCommand::Join(eth_addr, value, _) => {
                 if value.is_zero() {
                     result.expect_err("should not join with 0 value");
+                } else if pre_state.has_staked(&eth_addr) {
+                    result.expect_err("should not join again");
```

Comment:
If this changed you also have to update [join](https://github.com/consensus-shipyard/ipc-monorepo/blob/c4e8e0eb5db12fe0aa13b207af375e76b7e1cbb6/fendermint/fendermint/testing/contract-test/tests/staking/state.rs#L348-L351) in the state to be similar to [stake](https://github.com/consensus-shipyard/ipc-monorepo/blob/c4e8e0eb5db12fe0aa13b207af375e76b7e1cbb6/fendermint/fendermint/testing/contract-test/tests/staking/state.rs#L370-L371) and return if the validator has already joined.